plenum/bls/bls_key_register_pool_ledger.py

```diff
@@ -21,14 +26,17 @@ def _current_bls_keys(self):
         return self._load_keys_for_root()  # {node_name : BLS key}
 
     def get_key_by_name(self, node_name, pool_state_root_hash=None):
-        return self._current_bls_keys.get(node_name)
+        return self._current_bls_keys.get(node_name, None)
 
     def _load_keys_for_root(self):
         keys = {}
         for _, txn in self._ledger.getAllTxn():
             if get_type(txn) == NODE:
                 data = get_payload_data(txn)[DATA]
-                alias = data[ALIAS]
-                blskey = data.get(BLS_KEY)
-                keys[alias] = blskey
+                if not config.VALIDATE_BLS_SIGNATURE_WITHOUT_KEY_PROOF or \
```

Comment:
The condition needs to be `if not config.VALIDATE_BLS_SIGNATURE_WITHOUT_KEY_PROOF and data.get(BLS_KEY_PROOF, None) is None`